CONTRIBUTING.md

```diff
@@ -1,46 +1,90 @@
+* When you feel more conformable in your contributions, checkout [help wanted].
+* Advanced contributors can try to help with issues labeled [important] covering most relevant work at the time.
 
-Thanks for contributing!
+If any of aforementioned labels don't have unassigned issues, please [contact] one of the [maintainers] asking to triage more issues.
+
+[github issue tracker]: https://github.com/etcd-io/etcd/issues
+[good first issue]: https://github.com/etcd-io/etcd/labels/good%20first%20issue
+[help wanted]: https://github.com/etcd-io/etcd/labels/help%20wanted
+[maintainers]: https://github.com/etcd-io/etcd/blob/main/MAINTAINERS
+[important]: https://github.com/etcd-io/etcd/labels/important
+
+## Setup development environment
+
+etcd supported development environments include only linux-amd64.
+We strongly recommend using linux-amd64 for development as issues for non-supported environments will be ignored.
+Expanding list of supported requirements would require proper testing and support that is currently lacking in etcd project.
+If you want help etcd support your preferred environment, please [file an issue].
+
+Setup environment:
+- [Clone the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+- Install Go by following [installation](https://go.dev/doc/install). Please check minimal go version in [go.mod file](./go.mod#L3).
+- Install build tools (make):
+  - For ubuntu and debian run `sudo apt-get install build-essentials`
+- Verify that everything is installed by running `make build`
 
-### Code style
+[file an issue]: https://github.com/etcd-io/etcd/issues/new/choose
 
-The coding style suggested by the Golang community is used in etcd. See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details.
+## Implement your change
 
-Please follow this style to make etcd easy to review, maintain and develop.
+etcd code should follow coding style suggested by the Golang community.
+See the [style doc](https://github.com/golang/go/wiki/CodeReviewComments) for details.
 
-### Format of the commit message
+Please ensure that your change passes static analysis:
+- `make test-fmt` to format code.
+- `make lint` to run linter (requires [golangci-lint](https://golangci-lint.run/usage/install/))
+- `make lint-fix` to fix linter errors (requires [golangci-lint](https://golangci-lint.run/usage/install/))
 
-We follow a rough convention for commit messages that is designed to answer two
-questions: what changed and why. The subject line should feature the what and
-the body of the commit should describe the why.
+Please ensure that your change passes tests.
+- `make test-unit` to run unit tests.
+- `make test-integration` to run integration tests.
+- `make test-e2e` to run e2e tests.
 
+All changes are expected to come with unit test.
+All new features are expected to have either e2e or integration tests.
+
+# Commit your change
```

Comment:
`##` instead of one to keep it as the same level of the header ?

Reply:
+1